staging/src/k8s.io/client-go/tools/cache/shared_informer_test.go

```diff
@@ -357,3 +357,93 @@ func TestSharedInformerErrorHandling(t *testing.T) {
 	}
 	close(stop)
 }
+
+func TestSharedInformerRemoveHandler(t *testing.T) {
+	source := fcache.NewFakeControllerSource()
+	source.Add(&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod1"}})
+	source.ListError = fmt.Errorf("Access Denied")
+
+	informer := NewSharedInformer(source, &v1.Pod{}, 1*time.Second).(ExtendedSharedIndexInformer)
+
+	handler1 := &ResourceEventHandlerFuncs{}
+	informer.AddEventHandler(handler1)
+	handler2 := &ResourceEventHandlerFuncs{}
+	informer.AddEventHandler(handler2)
+
+	if informer.EventHandlerCount() == 0 {
```

Comment:
why not checking "informer.EventHandlerCount() != 2" to be precise

Reply:
yes, the test was originally intended for the bool HasEventHandler function before changing it according to a change request for this PR. I'll change this.